rskj-core/src/main/java/org/ethereum/vm/VM.java

```diff
@@ -1532,14 +1533,42 @@ private MessageCall getMessageCall(DataWord gas, DataWord codeAddress, Activatio
         // the callee will receive less gas than the parent expected.
         long calleeGas = Math.min(remainingGas, specifiedGasPlusMin);
 
+        // TODO:I next two lines can be skipped for RSKIP validation. Temp vars so RSKIP pseudocode var names match with the used ones. Later `remainingGas` and `gasCost` should be reused
+        long availableGas = remainingGas;
+        long standardCallCosts = gasCost;
+
         if (computeGas) {
-            gasCost = GasCost.add(gasCost, calleeGas);
-            spendOpCodeGas();
+            if (activations.isActive(RSKIP209)) {
+                program.spendGas(standardCallCosts, op.name());
+                long callGas = Math.min(specifiedGasPlusMin, availableGas); // TO_CHECK_1 @sergio to take into account transfer stipend, we cannot use `top-of-stack item` but `specifiedGasPlusMin`
```

Comment:
Yes, but I wonder why we were previously using specifiedGasPlusMin and not min(specifiedGas,minimumTransferGas) (instead of adding them).
I really can't understand why it was computed that way.

Reply:
callGas seems to hold the same value as calleeGas here. Why recompute ? It confuses the reader thinking both are different.

Reply:
> Yes, but I wonder why we were previously using specifiedGasPlusMin and not min(specifiedGas,minimumTransferGas) (instead of adding them).
> I really can't understand why it was computed that way.

I think we were incrementing the `specifiedGas` with the stipend in order to make the estimation bigger when dealing with a call with value, so as to not get OOG for these cases, as the stipend is needed during execution even if later refunded. Maybe [this comment](https://github.com/rsksmart/rskj/issues/948#issuecomment-520960211) provides some light.

---

> callGas seems to hold the same value as calleeGas here. Why recompute ? It confuses the reader thinking both are different.

Agree, left it just to match your pseudocode and ease the review, but there was a `TODO` to remove it